docs/proposal/20211210-launch-templates-managedmachinepools.md

```diff
@@ -0,0 +1,254 @@
+
+Cleaning up old versions of launch templates is currently handled by [PruneLaunchTemplateVersions](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/ec057ad6e613a6578f67bf68a6c77fbe772af933/pkg/cloud/services/ec2/launchtemplate.go#L265) which is sufficient for this change. We may want to make the minimum number of versions to keep configurable in the future but this can be covered by a different change.
+
+### Security Model
+
+There are no changes required to the security model. Access to the required CRDs is already declared for the controllers and as we are not adding any new kinds this doesn't need to change.
+
+No change is required to the AWS permissions the controller requires for reconciliation.
+
+### Risks and Mitigations
+
+The risk is that we are being constrained by the existing API definition used in unmanaged machine pools. This may raise unforeseen issues.
+
+## Alternatives
+
+### New `AWSLaunchTemplate` CRD & Controller
+
+The idea is that a `AWSLaunchTemplate` CRD would be created with an associated controller. The controller would then be responsible for reconciling the definition and managing the lifecycle of launch templates on AWS.
+
+#### Benefits
+
+* Single point of reconciliation and lifecycle management of launch templates in AWS. 
+* Separate lifecycle per launch template. So, we can change the number of previous instances to keep etc.
+
+#### Downsides
+
+* Additional complexity of orchestrating the creation of the launch template with the bootstrap data. The machine pool reconcilers would need to wait for the bootstrap data and the launch template.
+* Would require deprecation of fields in 2 CRDs (i.e both machine pool varieties). 
+
+#### Decision
+
+As `AWSMachinePool` already managed launch templates, it was felt that we should follow the same approach for consistency and it would be a smaller change.
+
+We can revisit the idea of a separate launch template kind in the future. The proposed change in this proposal will not preclude implementing this alternative in the future.
+
+## Upgrade Strategy
+
+The changes we are making to `AWSManagedMachinePool` are optional. Therefore, current users do not have to use the new `AWSLaunchTemplate` field. On upgrading there will be a new log entry written that informs the user that certain fields will be deprecated in the future.
+
+## Additional Details
+
+### Test Plan
+
+* There are currently no controller unit tests for the machine pools in CAPA. We do need to add tests, but this can be done as part of a separate change.
+* The EKS e2e tests will need to be updated so that a managed machine pool is created with a launch template specified.
+
+### Graduation Criteria
+
+With this proposal, we are planning to deprecate a number of fields on `AWSManagedMachinePool` and as this is a **beta level** API this means:
```

Comment:
Do we follow beta API criteria for experimental features?

Reply:
Thats a good question. I would say that we don't so i have updated the proposal.